File: sentimental-readability/readability.py
# TODO
#Readability in python

statement = input("Enter your statement: ")

letters = 0
words = 0
sentence = 0

for i in statement:
    if(i.isalpha()) == True:
        letters = letters + 1

    if (i.isspace()) == True:
        words = words + 1

    if (i == '.' or i == '?' or i == '!'):
        sentence = sentence + 1

words += 1
L = (letters * 100/words)
S = (sentence * 100/words)

#coleman - index to calculate the grade
index = round(0.0588 * (L) - 0.296 * (S) - 15.8)

if index > 16:
    print("Grade 16+")

elif index < 1:
    print("Before Grade 1")

else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <cs50.h> 
#include <stdio.h>
#include<string.h>
#include<ctype.h>
#include<math.h>

int main(void)
{ 
    float letters = 0;
    float words = 0;
    float sentences = 0;
    // Get an input from the user 
    string s = get_string("Text: ");
    
    // Find number of sentences, words and letters
    for(int i = 0,n = strlen(s); i<n; i++)
    {
        char ch = s[i];
        if(isalpha(ch))
        {
            letters++;
        }
       else if(isspace(ch))
        {
            words++;
        }
    
        if(ch == '.'|| ch == '!'|| ch =='?')
          {
            sentences++;
          }
    }
         
    words++;
// formula
int index = 0;
float X = (letters * 100.00/words);

float Y = (sentences * 100.00/words);

index = round(0.0588 * X - 0.296 * Y - 15.8);

// Result
    if(index > 16)
     {
      printf("Grade 16+\n");  
      }
  
  
     else if (index < 1)
      {
       printf("Before Grade 1\n");
      }
     else
      {
    printf("Grade %i\n", index);
       }
}
